File: ru.kmorozov.gbd.core.logger/src/main/java/ru/kmorozov/gbd/logger/events/AbstractEventSource.java
package ru.kmorozov.gbd.logger.events;

import ru.kmorozov.gbd.logger.progress.IProgress;

/**
 * Created by km on 25.12.2015.
 */
public abstract class AbstractEventSource implements IEventSource {

    private IProgress _processStatus;

    @Override
    public IProgress getProcessStatus() {
        return _processStatus;
    }

    protected void setProcessStatus(final IProgress processStatus) {
        this._processStatus = processStatus;
    }
}

File: ru.kmorozov.gbd.core.logger/src/main/java/ru/kmorozov/gbd/logger/events/AbstractEventSource.kt
package ru.kmorozov.gbd.logger.events

import ru.kmorozov.gbd.logger.progress.IProgress

/**
 * Created by km on 25.12.2015.
 */
abstract class AbstractEventSource : IEventSource {

    override var processStatus: IProgress
        get() = TODO("not implemented") //To change initializer of created properties use File | Settings | File Templates.
        set(value) {}
}
